pkg/k8s/api.go

```diff
@@ -245,6 +245,27 @@ func GetPodStatus(pod corev1.Pod) string {
 	return reason
 }
 
+// CheckForEnv checks for the given key in the given Container env variables
+func CheckForEnv(container *corev1.Container, envKey string) bool {
```

Comment:
Can we rename this method to `IsEnvPresent` or something similar?

Reply:
This is going to be removed once #5608 is merged as we only plan to check the newly added annotation by the injector.